include/NAS2D/MathUtils.h

```diff
@@ -29,5 +30,30 @@ namespace NAS2D
 
 	int divideUp(int to_divide, int divisor);
 
+	
+
+/**
+ * \fn T rangeMap(T a, T b)
+ *
+ * Applies a linear transform to map an input range to an output range for built-in types.
+ *
+ * \param	valueToMap: The value to be mapped from the input range to the output range. Must start in the closed range [minInputRange,maxInputRange]
+ * \param	minInputRange: Smallest value to map from.
+ * \param	maxInputRange: Largest value to map from.
+ * \param	minOutputRange: Smallest value to map to.
+ * \param	maxOutputRange: Largest value to map to.
+ *
+ * \return	Returns the mapped value in the closed range [minOutputRange,maxOutputRange]
+ */
+template<typename T>
+T rangeMap(const T& valueToMap, const T& minInputRange, const T& maxInputRange, const T& minOutputRange, const T& maxOutputRange)
+{
+	return (valueToMap - minInputRange) * (maxOutputRange - minOutputRange) / (maxInputRange - minInputRange) + minOutputRange;
+}
+
+Vector<float> rangeMap(const Vector<float>& valueToMap, const Vector<float>& minmaxInputRange, const Vector<float>& minmaxOutputRange);
+Vector<int> rangeMap(const Vector<int>& valueToMap, const Vector<int>& minmaxInputRange, const Vector<int>& minmaxOutputRange);
```

Comment:
Hah, I was actually thinking of suggesting something like this. I kind of stopped and realized though that `Vector` is perhaps not the right usage here. A `Vector` has independent `x` and `y` values, which represents different axis. Here we have `min` and `max` values, which are points along the same axis. This is probably an abuse of `Vector` and we should avoid re-using it here. A `Range` is a different concept from a `Vector`.

I would recommend getting rid of these methods for now. A later PR could upgrade the method with a `Range` type to provide such use.

----

Actually, the template method could potentially be applied to 2D spaces, mapping points between each other linearly. That would be different from the above, since you would have 2D points for both `min` and `max`, for both the domain and the range.